source/kraken/tests/apply_disruption_test.cpp

```diff
@@ -1161,3 +1161,499 @@ BOOST_AUTO_TEST_CASE(stop_point_deletion_test) {
     BOOST_CHECK_EQUAL(mvj_B->impacted_by.size(), 2);
 
 }
+
+BOOST_AUTO_TEST_CASE(add_simple_impact_on_line_section) {
+    ed::builder b("20160404");
+    b.sa("stop_area:1", 0, 0, false, true)("stop_point:10");
+    b.sa("stop_area:2", 0, 0, false, true)("stop_point:20");
+    b.sa("stop_area:3", 0, 0, false, true)("stop_point:30");
+    b.sa("stop_area:4", 0, 0, false, true)("stop_point:40");
+
+    b.vj("line:A", "111111", "", true, "vj:1")
+            ("stop_point:10", "08:10"_t, "08:11"_t)
+            ("stop_point:20", "08:20"_t, "08:21"_t)
+            ("stop_point:30", "08:30"_t, "08:31"_t)
+            ("stop_point:40", "08:40"_t, "08:41"_t);
+    b.vj("line:A", "101010", "", true, "vj:2")
+            ("stop_point:10", "09:10"_t, "09:11"_t)
+            ("stop_point:20", "09:20"_t, "09:21"_t)
+            ("stop_point:30", "09:30"_t, "09:31"_t)
+            ("stop_point:40", "09:40"_t, "09:41"_t);
+    b.vj("line:A", "001100", "", true, "vj:3")
+            ("stop_point:10", "10:10"_t, "10:11"_t)
+            ("stop_point:20", "10:20"_t, "10:21"_t)
+            ("stop_point:30", "10:30"_t, "10:31"_t)
+            ("stop_point:40", "10:40"_t, "10:41"_t);
+
+    b.generate_dummy_basis();
+    b.finish();
+    b.data->pt_data->index();
+    b.data->build_raptor();
+    b.data->build_uri();
+    b.data->meta->production_date = bg::date_period(bg::date(2016,4,4), bg::days(7));
+
+    navitia::apply_disruption(b.impact(nt::RTLevel::Adapted, "line_section_on_line:A_diverted")
+                              .severity(nt::disruption::Effect::NO_SERVICE)
+                              .on_line_section("line:A", "stop_area:2", "stop_area:3", {"line:A:0"})
+                              .application_periods(btp("20160404T120000"_dt, "20160406T090000"_dt))
+                              .get_disruption(),
+                              *b.data->pt_data, *b.data->meta);
+
+    auto vjmap = b.data->pt_data->vehicle_journeys_map;
+
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->lines.size(), 1);
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->routes.size(), 1);
+    // Only 5 vj, vj:3 shouldn't be affected
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->vehicle_journeys.size(), 5);
+
+    // Check the original vj
+    auto* vj = vjmap["vj:1"];
```

Comment:
minor: not 100% sure, but I don't think you need those map, I think you can access a vj by name with 
`b.get<VehicleJourney>("vj:1");`